dev/fattest.simplicity/src/componenttest/topology/impl/LibertyServer.java

```diff
@@ -450,10 +452,76 @@ public String getPathToAutoFVTNamedServer() {
     private final LogMonitor logMonitor;
 
     private boolean newLogsOnStart = FileLogHolder.NEW_LOGS_ON_START_DEFAULT;
+
+    public void setCheckpoint(Phase phase) {
+        checkpointInfo = new CheckPointInfo(phase);
+    }
+
     /**
-     * checkpointPhase defaults to null meaning no checkpoint taken
+     * When server.start is executed, perform a
+     *
+     * <pre>
+     * <code> "bin/server checkpoint --at==phase"</code>, followed by
+     * <code> "bin/server start"</code>
+     * </pre>
+     *
+     * @param phase
+     * @param autoRestor    if true initiate restore as part of serverStart
+     * @param beforeRestore beforeRestore lambda is called just before the server start
      */
-    private Phase checkpointPhase = null;
+    public void setCheckpoint(Phase phase, boolean autoRestore, Consumer<LibertyServer> beforeRestoreLambda) {
+        if (phase == null) {
```

Comment:
Why check for null here but not in the other setCheckpoint method above?  Why doesn't the simple `setCheckpoint(Phase phase)` method just call this one with `true` and a `null` consumer?

Reply:
Thanks. I moved the null check to constructor and redirected one parameter setCheckpoint method to call three parameter version.